examples/fluids/src/petsc_ops.c

```diff
@@ -213,17 +236,47 @@ PetscErrorCode VecCopyP2C(Vec X_petsc, CeedVector x_ceed) {
   PetscFunctionReturn(0);
 }
 
+/**
+ * @brief Create local PETSc Vecs for CeedOperator's active input/outputs
+ *
+ * @param[in]  dm     DM overwhich the Vecs would be used
+ * @param[in]  op     Operator to make the Vecs for
+ * @param[out] input  Vec for CeedOperator active input
+ * @param[out] output Vec for CeedOperator active output
+ */
+PetscErrorCode CeedOperatorCreateLocalVecs(DM dm, CeedOperator op, Vec *input, Vec *output) {
+  CeedSize input_size, output_size;
+  VecType  vec_type;
+  MPI_Comm comm = PetscObjectComm((PetscObject)dm);
+
+  PetscFunctionBeginUser;
+  CeedOperatorGetActiveVectorLengths(op, &input_size, &output_size);
+  PetscCall(DMGetVecType(dm, &vec_type));
+  if (input) {
+    PetscCall(VecCreate(comm, input));
```

Comment:
https://petsc.org/main/manualpages/DM/DMCreateLocalVector/

This would set the length and type for you. Unless the DM is just for getting the VecType?

Reply:
DM is just for getting the `VecType`. This is exclusively used for non-DM related `Vec`s. Though I can see how having a DM as an argument could confuse that. I thought about making it an input parameter, but that ends up adding an extra 3 lines to every single call to the function. I guess I could create a 

```c
VecType DMReturnVecType(DM dm) {
  VecType vec_type;
  DMGetVecType(dm, &vec_type);
  return vec_type;
}
```
Then do 
```c
PetscErrorCode CeedOperatorCreateLocalVecs(DM dm, CeedOperator op, Vec *input, Vec *output);

PetscCall(CeedOperatorCreateLocalVecs(DMReturnVecType(dm), op, &input, &output));
```

Reply:
I've pushed up changes to the above in fixup commits. If you're good with them, I'll squash them in.

Reply:
Will the DM that was previously passed ever not be of the right size? I suppose it might for some of the operators because the output might not actually have a DM associated with it?

Reply:
> Will the DM that was previously passed ever not be of the right size?

Yes. For example, the turbulent statistics are stored locally long-term at quadrature points, not nodes. So there isn't a `DM` that matches with that structure size.

> I suppose it might for some of the operators because the output might not actually have a DM associated with it?

For the turbulent statistics example, this is true for both inputs and outputs, though not at the same time. The collection phase takes in nodal solution field (associated with the 3D solution `DM`) and outputs the statistics at quadrature points (no associated `DM`).

Conversely, the processing step takes those statistics at quadrature points and computes the RHS of an L^2 projection, which is associated with the 2D statistics `DM`.